src/User/Handlers.hs

```diff
@@ -15,12 +21,70 @@ import IoInteractions
 handleUsers :: AppHandler
 handleUsers = do
   users <- selectAllUsers
-  renderWithSplices "users" $ userSplices $ map toUserPresenter users
+  renderWithSplices "users" $ usersSplices $ map toUserPresenter users
 
 handleUserAdd :: AppHandler
 handleUserAdd = do
-  sd <- liftIO getSystemDay
-  (view, muf) <- runForm "user_form" $ userForm sd
-  case muf of
-    Just uf -> insertUser uf >> redirect "/users"
-    Nothing -> heistLocal (bindDigestiveSplices view) $ render "users_add" 
\ No newline at end of file
+  currentDay <- liftIO getSystemDay
+  (view, muf) <- runForm "user_form" $ userForm currentDay
+  maybe (handleFormErrors view) insertUserIntoDb muf
+  where 
+    insertUserIntoDb uf = do 
+      createdUsers <- insertUser uf
+      bool userNotCreated (redirect "/users") (createdUsers>0)
+    userNotCreated = 
+      finishResponse 500 "Something went wrong while creating the user"
+    handleFormErrors view =
+      heistLocal (bindDigestiveSplices view) $ render "users_add" 
+
+handleUserEdit :: AppHandler
+handleUserEdit = do
+  key <- getUserKeyFromQueryString
+  user <- getUserFromDatabase key
+  currentDay <- liftIO getSystemDay
+  (view, muf) <- runForm "user_form" $ userForm currentDay
+  maybe (handleForm user view) (renderSuccess key) muf
+  where 
+    getUserKeyFromQueryString = do
+      mId <- getParam "id"
+      maybe badRequest (return . read . unpack) mId
+      where
+        badRequest = finishResponse 400 "Parameter 'id' was not found."
+    getUserFromDatabase key = do
+      recoveredUsers <- selectUserById key
+      maybe userNotFound return (S.headMay recoveredUsers)
+      where 
+        userNotFound = finishResponse 404 "User was not found."
```

Comment:
You don't have to always use the maybe function... is up to you when to use it... use it when it improves legibility, maybe, in this case, it would've been better to use a `case` expression such as
```hs
case S.headMay user of
  Nothing -> finishResponse 404 userNotFoundMsg
  Just u -> return u
```

or even

```hs
muser <- S.headMay <$> selectUserById key 
...
```
again, is up to you...